packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -2481,11 +2547,23 @@ def _update_round(self) -> None:
         Check whether the round has finished. If so, get the
         new round and set it as the current round.
         """
-        result: Optional[
+        # TODO: exit agent when termination is done ie, transaction_settlement_abci is finished after being scheduled
```

Comment:
Relating to my comment above, I have an idea how we can define this on the spec level via a transition from transaction_settlement_abci to a custom "shutdown" behaviour. This would be preferrable as we would not create any dependency on the transaction_settlement_abci. Can discuss Monday.